operator/pkg/helmreconciler/apply.go

```diff
@@ -0,0 +1,203 @@
+// Copyright 2020 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package helmreconciler
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"helm.sh/helm/v3/pkg/releaseutil"
+	errors2 "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+	util2 "k8s.io/kubectl/pkg/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"istio.io/istio/operator/pkg/cache"
+	"istio.io/istio/operator/pkg/name"
+	"istio.io/istio/operator/pkg/object"
+	"istio.io/istio/operator/pkg/util"
+)
+
+// ApplyManifest applies the manifest to create or update resources. It returns the processed (created or updated)
+// objects and the number of objects in the manifests.
+// waitUnitReady waits until all resources in the manifest are in a ready state.
+func (h *HelmReconciler) ApplyManifest(manifest releaseutil.Manifest, waitUntilReady bool) (object.K8sObjects, int, error) {
```

Comment:
nit: can we keep the original function name here? there are multiple applymanifest along the path

Reply:
processManifest seems incorrect because we are actually applying the manifest in the function. though we have multiple applyManifests, they are all in fact doing what they say they are doing. the fact that we have multiple of these could point to code duplication so perhaps that's the issue here.